README.md

```diff
@@ -57,6 +57,29 @@ To deal with this:
 Additionally, :focusring matches non-native elements as if they were
 native button elements.
 
+### Note: Styling non-native elements which should always match `focus-ring`
+
+This is not currently part of the spec,
+but a mechanism is needed to explain the ability of native text fields
+to match `:focus-ring` regardless of how focus arrived on the element.
+
+Two different strategies may enable this
+("`-bikeshed`" placeholder indicates that these names are by no means final!):
+
+1. Add a new keyword value to the outline shorthand that represents whatever the default UA `::focus-ring` is. Then authors can do:
+
+        :focus {
+            outline: platform-default-focus-outline-style-bikeshed;
+        }
+
+2. Add a new CSS property that controls "keyboard modality" vs non-"keyboard modality" behavior, e.g.
+
+        my-custom-texty-element {
+            should-show-focus-on-mouse-click-bikeshed: on-bikeshed;
```

Comment:
OOOOOOOOH. got it. yes that would be nice. I think the way the use case was pitched didn't click in my head... also it's 1am

You may be able to rephrase it as:

```
In addition to `:focus-ring`, there are additional, speculative, missing primitives which may make working with focus easier.

1. An element should have a way to restore the UA default outline. Because each UA has a different default style, there's no easy way to restore the outline if a prior selector has overridden it (short of UA sniffing).

2. An element should have a way to express when it should display a focus ring. On keyboard focus, on mouse focus, or on both.

These could help explain the behavior of native text fields...
```